data/fs/fs.go

```diff
@@ -0,0 +1,176 @@
+package fs
+
+import (
+	"log"
+	"os"
+	"path"
+	"skrive/data"
+	"sort"
+	"strings"
+
+	gonanoid "github.com/matoous/go-nanoid/v2"
+)
+
+type FsStorage struct {
+	Path string
+}
+
+func GetPath(fileArgument string) (*string, error) {
+	if len(fileArgument) > 0 {
+		return &fileArgument, nil
+	}
+
+	if value, isDefined := os.LookupEnv("SKRIVE_DOSES_PATH"); isDefined {
+		return &value, nil
+	}
+
+	if dirname, err := os.UserHomeDir(); err != nil {
+		return nil, homePathError{}
+	} else {
+		var p = path.Join(dirname, "doses.dat")
+		return &p, nil
+	}
+}
+
+type homePathError struct{}
+
+func (e homePathError) Error() string {
+	return "Could not find home directory"
```

Comment:
It's a best practice to write error messages as whole sentences with punctuation.
```suggestion
	return "Could not find a home directory."
```